routes/web.php

```diff
@@ -30,6 +30,7 @@
         ->namespace('Teacher')
         ->group(static function (): void {
             Route::get('/', 'Dashboard')->name('teacher-dashboard');
+            Route::get('/student/{student}', 'StudentDetails')->name('teacher-student-details');
```

Comment:
There needs to be some authorization on this route. It should not be possible that any teacher can access the details of any student, we need to make sure that the teacher **has actually been assigned** to the given student. What I would suggest for this is using Laravel's policies. There are some other policies that you can open to see how they're implemented or follow the [Laravel docs](https://laravel.com/docs/5.7/authorization#creating-policies). You can then create a method on the policy like  
```php
public function isAssignedToStudent(Student $teacher, Student $student): bool;
```
We can then use this policy on any place within the application, for example on [routes](https://laravel.com/docs/5.7/authorization#via-middleware).